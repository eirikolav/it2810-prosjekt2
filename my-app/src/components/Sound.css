.audiocontainer {
    display: flex;
    flex-direction: column;
    width: 300px;
    margin-bottom: 5px;
    margin-top: 45px;
    
}

.audiobutton {
    margin-left: 75px;
}

@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 812px)  { 

    .audiocontainer{
        margin-left: 60px;
    }

    .audiobutton {
        margin-left: 50px;
    }
}


