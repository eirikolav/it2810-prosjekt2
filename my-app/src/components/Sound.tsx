import React from 'react';
import './Sound.css'


interface SoundProps {
    //Interface to types of props passed to Sound
    soundHandleClick: () => void;
    index: number;
    sound: string;
}

export default class Sound extends React.Component<SoundProps, {}> {
    //Class to render the audio-object. Uses props from App.tsx to display pictures
    //Button to change the song.  
    render() {
        return (
            <div className="audiocontainer">
                <div>
                    <audio src={this.props.sound} controls autoPlay></audio>
                </div>
                <div className="audiobutton">
                    <button onClick={this.props.soundHandleClick}>Bytt lyd</button>
                </div>

            </div>
        )
    }

}
