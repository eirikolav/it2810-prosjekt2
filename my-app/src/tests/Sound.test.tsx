import React from 'react'
import renderer from 'react-test-renderer'
import Sound from '../components/Sound'

test("Test Sound", () => {
    const tree = renderer.create(<Sound />).toJSON()
    expect(tree).toMatchSnapshot()
})