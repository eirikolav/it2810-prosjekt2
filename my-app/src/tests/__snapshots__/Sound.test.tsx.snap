// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sound 1`] = `
<div
  className="audiocontainer"
>
  <div>
    <audio
      autoPlay={true}
      controls={true}
    />
  </div>
  <div
    className="audiobutton"
  >
    <button>
      Bytt lyd
    </button>
  </div>
</div>
`;
